sqlglot/optimizer/simplify.py

```diff
@@ -1097,15 +1119,25 @@ def extract_interval(expression):
         return None
 
 
-def date_literal(date):
-    return exp.cast(
-        exp.Literal.string(date),
-        (
+def extract_type(*expressions):
+    target_type = None
+    for expression in expressions:
+        target_type = expression.to if isinstance(expression, exp.Cast) else expression.type
+        if target_type:
```

Comment:
What is target_type is UNKNOWN?

Reply:
It should be ignored in `date_literal` using the `TEMPORAL_TYPES` check.